packages/widget.api/src/config.ts

```diff
@@ -26,9 +26,21 @@ interface ArrayParameterDefinition<S extends ParameterValue.PrimitiveType> {
   displayName: string;
   subType: S;
 }
+interface ObjectSetParameterDefinition<T extends ObjectType> {
+  type: "objectSet";
+  displayName: string;
+  objectType: T;
+}
 export type ParameterDefinition =
   | PrimitiveParameterDefinition<ParameterValue.PrimitiveType>
-  | ArrayParameterDefinition<ParameterValue.PrimitiveType>;
+  | ArrayParameterDefinition<ParameterValue.PrimitiveType>
+  | ObjectSetParameterDefinition<ObjectType>;
+
+export type ManifestParameterDefinition =
+  | Exclude<ParameterDefinition, ObjectSetParameterDefinition<ObjectType>>
+  | (Omit<ObjectSetParameterDefinition<ObjectType>, "objectType">) & {
+    objectTypeRids: [string];
```

Comment:
Do you prefer to keep the manifest format array as well or start it off as string? Right now we have:
- Config format single object type
- Manifest format array
- Stored format array

For the manifest we have some flexibility to change it like allowing `string | string[]` in the future but starting it off as array is fine as well if it's clear that it will be enforced to be non-empty with one rid to start with per your types ([string])

Reply:
I'd like to keep it `string[]` in case we expand in the future. I think the backend change could look ugly if we have to start supporting `string | string[]` in the types? +1 on validating that it's a `[string]` for now though!